File: frontend/src/app/shared/service/admin/price/PriceAdmin.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

import {environment} from '../../../../../environments/environment';

import {PriceDto} from '../../../model/price/Price.model';
import {PriceCriteria} from '../../../criteria/price/PriceCriteria.model';
import {AbstractService} from "../../../../zynerator/service/AbstractService";


@Injectable({
    providedIn: 'root'
})
export class PriceAdminService extends AbstractService<PriceDto, PriceCriteria> {
    constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    get API() {
        return environment.apiUrlAlcservice + 'admin/price/';
    }

    public constrcutDto(): PriceDto {
        return new PriceDto();
    }

    public constrcutCriteria(): PriceCriteria {
        return new PriceCriteria();
    }
}

File: backend/src/main/java/ma/zs/alc/service/facade/admin/price/PriceAdminService.java
package ma.zs.alc.service.facade.admin.price;

import java.util.List;
import ma.zs.alc.bean.core.price.Price;
import ma.zs.alc.dao.criteria.core.price.PriceCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface PriceAdminService extends  IService<Price,PriceCriteria>  {

    List<Price> findByCollaboratorId(Long id);
    int deleteByCollaboratorId(Long id);
    long countByCollaboratorId(Long id);




}
